### PR TITLE
Accept exp and nbf in ms

### PR DESCRIPTION
**Pull request by solisoft**
Some JWT tokens contains time fields in ms instead of seconds and it raise an error.

This PR just convert time in seconds if needed and validate then the token

### CODE REVIEW

File: examples/basic_usage.cr [hunk 1]
```diff
@@ -9,3 +9,8 @@ pp token
 payload, header = JWT.decode(token, "SecretKey", "HS256")
 pp payload
 pp header
+
+# exp field in ms
+payload, header = JWT.decode("eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzUxMiJ9.eyJ1aWQiOiI4MjY4OTEwOSIsImlhdCI6MTUwMzM5NDAwNjI1NSwicGF5bG9hZCI6eyJyb2xlIjoiRWxldmUiLCJlbWFpbCI6ImVsZXZlMUBlc3NhaS5jb20iLCJ1c2VybmFtZSI6IlAuQkFSUkVTIn0sImV4cCI6MTUwMzM5NzYwNjI1OH0.FcfRg1JYqIT9ZQQneEDRbt16_ltN6dyAfuFezsN0uBJ8jO-jAPzUYqlRJ77hKvjF2h51jvdtII5GVoEw72TIyw", "fJShJ9ZH0c05eXZNzi0qcBV3WeLar2Vi9zIdo8Ggbk3LoLTpkzTef8L0AYlwzcO1YotNsaBydIYk50F2", "HS512")
+pp payload
+pp header
```
Comment (veelenga) [inline examples/basic_usage.cr:16]: Your example does not work for me:

```sh
$ cr examples/basic_usage.cr                                                                                                                                                     
token # => "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJmb28iOiJiYXIifQ.Y3shN5Wh4FmOPM34biIm9QQmat373hJFKNxgSANQWJo"
payload # => {"foo" => "bar"}
header # => {"typ" => "JWT", "alg" => "HS256"}
Signature is expired (JWT::ExpiredSignatureError)
0x1028dd6c5: *CallStack::unwind:Array(Pointer(Void)) at ??
0x1028dd661: *CallStack#initialize:Array(Pointer(Void)) at ??
0x1028dd638: *CallStack::new:CallStack at ??
0x1028dc495: *raise<JWT::ExpiredSignatureError>:NoReturn at ??
0x102936e38: *JWT@JWT::validate_exp!<(Array(JSON::Type) | Bool | Float64 | Hash(String, JSON::Type) | Int64 | String | Nil)>:Nil at ??
0x102936aac: *JWT@JWT::decode<String, String, String>:Tuple(Hash(String, JSON::Type), Hash(String, JSON::Type)) at ??
0x1028c7857: __crystal_main at ??
0x1028dafc8: main at ??

```

**Comment by solisoft** [comment]
Well this is because the token has expire :)

**Comment by solisoft** [comment]
But at least it stop raising the int32 error

**Comment by veelenga** [comment]
@solisoft yes, but that means this example is just useless. Could you please add two tests instead for the following cases:

```crystal
# positive case
token = JWT.encode({"foo" => "bar", "exp" => Time.now.epoch + 1}, "SecretKey", "HS256")
JWT.decode(token, "SecretKey", "HS256") # payload is successfully decoded

# negative case
token = JWT.encode({"foo" => "bar", "exp" => Time.now.epoch - 1}, "SecretKey", "HS256")
JWT.decode(token, "SecretKey", "HS256") # JWT::ExpiredSignatureError is raised
```

**Review by veelenga — CHANGES_REQUESTED** [review]
@solisoft yes, but that means this example is just useless. Could you please add specs instead for the following cases:

```crystal
# positive case
token = JWT.encode({"foo" => "bar", "exp" => Time.now.epoch_ms + 1000}, "SecretKey", "HS256")
JWT.decode(token, "SecretKey", "HS256") # payload is successfully decoded

# negative case
token = JWT.encode({"foo" => "bar", "exp" => Time.now.epoch_ms - 1000}, "SecretKey", "HS256")
JWT.decode(token, "SecretKey", "HS256") # JWT::ExpiredSignatureError is raised
```

**Comment by solisoft** [comment]
Ok Actually I think we can remove this PR because it's a bug from arangodb token generator ... not this lib.
https://github.com/arangodb/arangodb/issues/2249

**Comment by veelenga** [comment]
Hm, okay, so only seconds should be supported... Thank you!

**Pull request closed without merging**